File: Vanta Sppech Android/app/src/main/java/com/vanta/speech/core/eas/model/EASAttendee.kt
package com.vanta.speech.core.eas.model

import kotlinx.serialization.Serializable

/**
 * Meeting attendee from EAS calendar event
 */
@Serializable
data class EASAttendee(
    /** Email address */
    val email: String,

    /** Display name */
    val name: String,

    /** Attendee type (required, optional, resource) */
    val type: AttendeeType = AttendeeType.REQUIRED,

    /** Response status (accepted, declined, etc.) */
    val status: ResponseStatus? = null
)

/**
 * Attendee type values from EAS protocol
 */
enum class AttendeeType(val value: Int) {
    REQUIRED(1),
    OPTIONAL(2),
    RESOURCE(3);

    val displayName: String
        get() = when (this) {
            REQUIRED -> "Обязательный"
            OPTIONAL -> "Необязательный"
            RESOURCE -> "Ресурс"
        }

    companion object {
        fun fromValue(value: Int): AttendeeType {
            return entries.find { it.value == value } ?: REQUIRED
        }
    }
}

/**
 * Response status values from EAS protocol
 */
enum class ResponseStatus(val value: Int) {
    NONE(0),
    ORGANIZER(1),
    TENTATIVE(2),
    ACCEPTED(3),
    DECLINED(4),
    NOT_RESPONDED(5);

    val displayName: String
        get() = when (this) {
            NONE -> "Нет ответа"
            ORGANIZER -> "Организатор"
            TENTATIVE -> "Возможно"
            ACCEPTED -> "Принято"
            DECLINED -> "Отклонено"
            NOT_RESPONDED -> "Не ответил"
        }

    val iconName: String
        get() = when (this) {
            NONE, NOT_RESPONDED -> "help_outline"
            ORGANIZER -> "star"
            TENTATIVE -> "help"
            ACCEPTED -> "check_circle"
            DECLINED -> "cancel"
        }

    companion object {
        fun fromValue(value: Int): ResponseStatus? {
            return entries.find { it.value == value }
        }
    }
}

File: Vanta Speech iOS/Vanta Speech/Core/EAS/Models/EASAttendee.swift
import Foundation

/// Meeting attendee from EAS calendar event
struct EASAttendee: Codable, Equatable, Identifiable {
    /// Email address (used as ID)
    let email: String

    /// Display name
    let name: String

    /// Attendee type (required, optional, resource)
    let type: EASAttendeeType

    /// Response status (accepted, declined, etc.)
    let status: EASResponseStatus?

    var id: String { email }

    // MARK: - Initialization

    init(email: String, name: String, type: EASAttendeeType = .required, status: EASResponseStatus? = nil) {
        self.email = email
        self.name = name
        self.type = type
        self.status = status
    }
}

/// Attendee type values from EAS protocol
enum EASAttendeeType: Int, Codable, Equatable {
    case required = 1
    case optional = 2
    case resource = 3

    var displayName: String {
        switch self {
        case .required: return "Обязательный"
        case .optional: return "Необязательный"
        case .resource: return "Ресурс"
        }
    }
}

/// Response status values from EAS protocol
enum EASResponseStatus: Int, Codable, Equatable {
    case none = 0
    case organizer = 1
    case tentative = 2
    case accepted = 3
    case declined = 4
    case notResponded = 5

    var displayName: String {
        switch self {
        case .none: return "Нет ответа"
        case .organizer: return "Организатор"
        case .tentative: return "Возможно"
        case .accepted: return "Принято"
        case .declined: return "Отклонено"
        case .notResponded: return "Не ответил"
        }
    }

    var iconName: String {
        switch self {
        case .none, .notResponded: return "questionmark.circle"
        case .organizer: return "star.circle.fill"
        case .tentative: return "questionmark.circle.fill"
        case .accepted: return "checkmark.circle.fill"
        case .declined: return "xmark.circle.fill"
        }
    }
}
